### PR TITLE
add unit test for big decimal in map

### PR DESCRIPTION
**Pull request by wenxuwan**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by wongoo** [comment]
@wenxuwan  try add the following code in serialize_test.go

```go

func TestDecimalMapJavaDecode(t *testing.T) {
	var d big.Decimal
	_ = d.FromString("100.256")
	d.Value = d.String()
	m := map[interface{}]interface{}{"test_BigDecimal": &d}

	out, err := decodeJavaResponse("customReplyTypedFixedDecimalMap", "", false)
	if err != nil {
		t.Errorf("%#v %v", out, err)
		return
	}

	resp := out.(map[interface{}]interface{})
	assert.True(t, reflect.DeepEqual(m, resp))
}
```

File: go.mod [hunk 1]
```diff
@@ -6,3 +6,4 @@ require (
 	github.com/stretchr/testify v1.4.0
 )
 
+go 1.13
```
Comment (wongoo) [inline go.mod:9]: not add go version

File: map.go [hunk 2]
```diff
@@ -245,12 +245,12 @@ func (d *Decoder) decMap(flag int32) (interface{}, error) {
 	case tag == BC_REF:
 		return d.decRef(int32(tag))
 	case tag == BC_MAP:
-		if t, err = d.decType(); err != nil {
```
Comment (wongoo) [inline map.go:248]: should reset

**Comment by wongoo** [comment]
@wenxuwan u can close this issue now for that fixed in https://github.com/apache/dubbo-go-hessian2/pull/261

**Comment by wenxuwan** [comment]
> @wenxuwan u can close this issue now for that fixed in #261

ok ,close now

**Pull request closed without merging**